python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -195,12 +241,60 @@ async def complete_prompt(
                 top_p=template.config.completion.top_p,
                 temperature=template.config.completion.temperature,
                 max_tokens=max_tokens,
+                tools=tools,
+                tool_choice=tool_choice,
+                parallel_tool_calls=parallel_tool_calls,
                 extra_body=extra_body,
             )
 
             if self._options.logger is not None:
                 self._options.logger.debug("COMPLETION:\n%s", completion.model_dump_json())
 
+            # Handle tools flow
+            action_calls: List[ActionCall] = []
+            response_message = completion.choices[0].message
+
+            if is_tools_aug:
+                tool_calls = response_message.tool_calls
+                if tool_calls and len(tool_calls) > 0:
+                    if not parallel_tool_calls and len(tool_calls) > 1:
```

Comment:
you don't need to validate the responses from the openai service, you can assume the service will return valid responses.